server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -1364,29 +1377,42 @@ private void initializeRunningWorkers() {
             LOGGER.trace("In initializeRunningWorkers for Job {}", jobId);
             // publish a refresh before enqueuing tasks to the Scheduler, as there is a potential race between
             // WorkerRegistryV2 getting updated and isWorkerValid being called from SchedulingService loop
-            // If worker is not found in the SchedulingService loop, it is conisdered invalid and prematurely
-            // removed from Fenzo state
+            // If worker is not found in the SchedulingService loop, it is considered invalid and prematurely
+            // removed from Fenzo state.
             markStageAssignmentsChanged(true);
 
-            for (IMantisStageMetadata stageMeta : mantisJobMetaData.getStageMetadata().values()) {
+            List<JobWorker> workersToResubmit = new ArrayList<>();
 
+            for (IMantisStageMetadata stageMeta : mantisJobMetaData.getStageMetadata().values()) {
                 Map<Integer, WorkerHost> workerHosts = new HashMap<>();
+
                 for (JobWorker worker : stageMeta.getAllWorkers()) {
                     IMantisWorkerMetadata wm = worker.getMetadata();
+
+                    Optional<WorkerPorts> workerPortsOptional = wm.getPorts();
+                    if (!workerPortsOptional.isPresent() || !workerPortsOptional.get().isValid()) {
+                        numMissingWorkerPorts.increment();
+                        // Mark these workers with missing ports for resubmission.
+                        workersToResubmit.add(worker);
```

Comment:
this would also resubmit Accepted state workers. The worker ports validation is only applicable to running workers.

Reply:
I see. I'll move this into the block below `WorkerState.isRunningState`.